src/panic.rs

```diff
@@ -11,3 +13,16 @@ pyo3_exception!(
     PanicException,
     PyBaseException
 );
+
+impl PanicException {
+    // Try to format the error in the same way panic does
+    pub(crate) fn from_panic(e: Box<dyn Any + Send + 'static>) -> PyErr {
```

Comment:
`std::panic::PanicInfo` calls this the "payload", so I think we can do similar:

```suggestion
    pub(crate) fn from_panic_payload(payload: &(dyn Any + Send + 'static)) -> PyErr {
```